x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/FieldValueFetcher.java

```diff
@@ -69,27 +71,39 @@ Object format(Object value) {
         }
     }
 
+    private static Map<String, FieldValueFetcher> build(SearchExecutionContext context, String field) {
+        MappedFieldType fieldType = context.getFieldType(field);
+        assert fieldType != null : "Unknown field type for field: [" + field + "]";
+        Map<String, FieldValueFetcher> fetchers = new HashMap<>();
+
+        if (fieldType instanceof AggregateDoubleMetricFieldMapper.AggregateDoubleMetricFieldType aggMetricFieldType) {
```

Comment:
Is this one of the threads you'd pull on to fix the other `instanceof` checks? If the field type built the fetcher - or if we had a `Map<Class<FieldType>, FetcherFactory>` or something?